core/test/base/batch_lin_op.cpp

```diff
@@ -0,0 +1,164 @@
+                                     gko::batch_dim<2>(1, gko::dim<2>{3, 5}))},
+          op2{DummyBatchLinOp::create(ref2,
+                                      gko::batch_dim<2>(2, gko::dim<2>{3, 5}))}
+    {}
+
+    std::shared_ptr<const gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::ReferenceExecutor> ref2;
+    std::unique_ptr<DummyBatchLinOp> op;
+    std::unique_ptr<DummyBatchLinOp> op2;
+};
+
+
+TEST_F(EnableBatchLinOp, KnowsNumBatchItems)
+{
+    ASSERT_EQ(op->get_num_batch_items(), 1);
+    ASSERT_EQ(op2->get_num_batch_items(), 2);
+}
+
+
+TEST_F(EnableBatchLinOp, KnowsItsSizes)
+{
+    auto op1_sizes = gko::batch_dim<2>(1, gko::dim<2>{3, 5});
+    auto op2_sizes = gko::batch_dim<2>(2, gko::dim<2>{3, 5});
+    ASSERT_EQ(op->get_size(), op1_sizes);
+    ASSERT_EQ(op2->get_size(), op2_sizes);
+}
+
+
+template <typename T = int>
+class DummyBatchLinOpWithFactory
+    : public gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory<T>> {
+public:
+    DummyBatchLinOpWithFactory(std::shared_ptr<const gko::Executor> exec)
+        : gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory>(exec)
+    {}
+
+    GKO_CREATE_FACTORY_PARAMETERS(parameters, Factory)
+    {
+        T GKO_FACTORY_PARAMETER_SCALAR(value, T{5});
+    };
+    GKO_ENABLE_BATCH_LIN_OP_FACTORY(DummyBatchLinOpWithFactory, parameters,
+                                    Factory);
+    GKO_ENABLE_BUILD_METHOD(Factory);
+
+    DummyBatchLinOpWithFactory(const Factory* factory,
+                               std::shared_ptr<const gko::batch::BatchLinOp> op)
+        : gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory>(
+              factory->get_executor()),
+          parameters_{factory->get_parameters()},
+          op_{op}
+    {}
+
+    std::shared_ptr<const gko::batch::BatchLinOp> op_;
+};
+
+
+class EnableBatchLinOpFactory : public ::testing::Test {
+protected:
+    EnableBatchLinOpFactory() : ref{gko::ReferenceExecutor::create()} {}
+
+    std::shared_ptr<const gko::ReferenceExecutor> ref;
+};
+
+
+TEST_F(EnableBatchLinOpFactory, CreatesDefaultFactory)
+{
+    auto factory = DummyBatchLinOpWithFactory<>::build().on(ref);
+
+    ASSERT_EQ(factory->get_parameters().value, 5);
+    ASSERT_EQ(factory->get_executor(), ref);
+}
+
+
+TEST_F(EnableBatchLinOpFactory, CreatesFactoryWithParameters)
+{
+    auto factory = DummyBatchLinOpWithFactory<>::build().with_value(7).on(ref);
+
+    ASSERT_EQ(factory->get_parameters().value, 7);
+    ASSERT_EQ(factory->get_executor(), ref);
+}
+
+
+TEST_F(EnableBatchLinOpFactory, PassesParametersToBatchLinOp)
+{
+    auto dummy = gko::share(
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 5})));
+    auto factory = DummyBatchLinOpWithFactory<>::build().with_value(6).on(ref);
+
+    auto op = factory->generate(dummy);
+
+    ASSERT_EQ(op->get_executor(), ref);
+    ASSERT_EQ(op->get_parameters().value, 6);
+    ASSERT_EQ(op->op_.get(), dummy.get());
+}
+
```

Comment:
miss FactoryGenerateIsLogged, CopiesLinOpToOtherExecutor testing because BatchLinOpFactory has the generate function and logger.